### PR TITLE
Remove periods from the end of package descriptions

### PR DESCRIPTION
**Pull request by siraben**
###### Motivation for this change
Addresses #97685

###### Things done
Got the list of packages to fix with a Nix script (see #97685), then went over the paths with an `awk` file;

```awk
{ system("sed --in-place='' -re 's#(description\\s*=\\s*\".+)\\.\"#\\1\"#' " $0)}
```

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/haskell-modules/hackage-packages.nix [hunk 1]
```diff
@@ -1041,7 +1041,7 @@ self: {
        libraryHaskellDepends = [
          base bytestring mtl process template-haskell
        ];
-       description = "A simple music library with the capability of generating .ly and .mid files.";
```
Comment (cdepillabout) [inline pkgs/development/haskell-modules/hackage-packages.nix:1044]: This file is automatically generated, so these changes would just get overwritten next time it is generated.

These descriptions are taken directly from the upstream packages.  If you wanted to fix this, you'd have to submit a PR to https://github.com/NixOS/cabal2nix (which is the tool that does the generations).

**Comment by teto** [comment]
doesn't mean much if we can't enforce this rule via ofborg. Also some of these descriptions are generated so we would need to fix those descriptions as well. Not worth the hassle IMO.

**Comment by siraben** [comment]
We can put this in ofborg (see #97685), but for now, it would be good to fix up the description of hundreds of packages.

**Comment by siraben** [comment]
I'll remove the edits to the autogenerated packages.

**Comment by siraben** [comment]
@worldofpeace Should I resolve the merge conflicts and have this be merged? The plan is for @GrahamcOfBorg to check this for new PRs in the future.

**Comment by worldofpeace** [comment]
Sure, let's try to merge it quickly to avoid any more conflicts.

**Comment by worldofpeace** [comment]
> @worldofpeace Should I resolve the merge conflicts and have this be merged? The plan is for @GrahamcOfBorg to check this for new PRs in the future.

I think we're actually considering moving everything into github checks.

**Comment by jonringer** [comment]
> I think we're actually considering moving everything into github checks.

I'm okay with this, scales much better; and don't have to bug graham all the time

**Comment by siraben** [comment]
For future reference,
<details>
<summary>Nix script to fetch paths to derivations whose descriptions end with period</summary>

```nix
let
  pkgs = import ./../../default.nix {};

  safeEval = e: let result = builtins.tryEval e;
                in
                  if result.success
                  then result.value
                  else [ ];

  packagesWith = cond:
    pkgs.lib.concatLists (pkgs.lib.mapAttrsToList
      (name: pkg: safeEval
        (if pkgs.lib.isDerivation pkg && cond name pkg
         then [ (builtins.elemAt (pkgs.lib.splitString ":" pkg.meta.position) 0) ]
         else [ ]))
      pkgs);

  hasDesc = pkg: builtins.hasAttr "description" pkg.meta;
  lastChar = x: builtins.substring (builtins.stringLength x - 1) (-1) x;
  report = desc: l:
    builtins.trace (pkgs.lib.concatStringsSep "\n" l) "";
in
{
  periodDesc =
    report "have a period at the end of their description"
      (packagesWith
        (name: pkg:
          if hasDesc pkg && builtins.stringLength pkg.meta.description != 0
          then lastChar pkg.meta.description == "."
          else false));
}
```
</details>

The output of the script is fed into an awk file with contents
```awk
{ system("sed --in-place='' -re 's#(description\\s*=\\s*\".+)\\.\"#\\1\"#' " $0)}
```

Finally, I went through the changed files to see which ones had a delta greater than 2, indicating it was probably a package collection (Haskell/Node/Lua packages came up).

**Comment by siraben** [comment]
@jonringer @worldofpeace Should be ready to merge.

File: pkgs/applications/science/math/bliss/default.nix [hunk 2]
```diff
@@ -22,14 +22,14 @@ stdenv.mkDerivation rec {
 
   installPhase = ''
     mkdir -p $out/bin $out/share/doc/bliss $out/lib $out/include/bliss
-    mv bliss $out/bin 
+    mv bliss $out/bin
```
Comment (jonringer) [inline pkgs/applications/science/math/bliss/default.nix:25]: oh, the rebuild is due to whitespace, for the check.

**Review by jonringer — APPROVED** [review]
LGTM

Result of nixpkgs-review pr 97835 [1](https://github.com/Mic92/nixpkgs-review)<details>  <summary>1 package blacklisted:</summary>  <ul>    <li>tests.nixos-functions.nixos-test</li>  </ul></details><details>  <summary>4 packages built:</summary>  <ul>    <li>bliss</li>    <li>documentation-highlighter</li>    <li>polymake</li>    <li>trilium-desktop</li>  </ul></details>

**Pull request merged**